uktrade/data-hub-frontend
# Feature/create investment opportunity

## Description of change

Adds the _Create UK investment opportumity_ page at `/investments/opportunities/create`.

Note the form is using the vanila `<form/>` instead of any of the `Form` components that we have, all of which suffer from a [bug where they can't be used in conjunction with the `Task` component](https://trello.com/c/C0qlxm1N/542-fix-the-flawed-isloading-mechanism-of-forms).

This PR also introduces three new components: `HardRedirect`, `ReferrerLink` and `TaskLoadingBox`.

## Test instructions

1. Navigate to `/investments/opportunities/create`
2. You should see the _Create UK investment opportumity_ form
3. Submit the form without entering any value into the _ Opportunity name_ field
4. You should see a validation error saying "Enter an opportunity name"
5. Now fill out the field with some text
6. Submit the form
7. Depending on what version of the API you are running it (whether a change was incorporated to the `/v4/large-capital-opportunity` which reduces the number of required fields to the POST method to only the `name`) it should
  - Render a `Task` error message overlay on top of the form
  - Redirect to `/investments/opportunities/<id-of-the-newly-created-opportunity>`

## Screenshots

<img width="839" alt="Screen Shot 2021-03-30 at 1 56 29 PM" src="https://user-images.githubusercontent.com/2333157/112992127-cea87800-915f-11eb-91cf-4f80fb7b3036.png">
